Documentation: Add guide page about Marketplace migration (#312)
cloudamqp/terraform-provider-cloudamqp: docs/guides/info_marketplace_migration.md

---
layout: "cloudamqp"
page_title: "Marketplace migration"
subcategory: "info"
description: |-
  Update resource and dependencies identifiers.
---

# Marketplace migration

> When migrating from our marketplace apps to CloudAMQP. The resource identifiers will be updated.
> Below follows an example of how to fetch the new identifiers and which needs to be updated.

Reviewer: ```suggestion
If you migrate your CloudAMQP billing from a cloud marketplace to directly with CloudAMQP (or vice versa), the resource identifiers will change. Below follows an example on how to fetch the new identifiers and update them.
```

## Configuration file

Basic example of a standalone managed VPC, CloudAMQP instance and firewall.

```hcl
resource "cloudamqp_vpc" "vpc" {
  name = "instance"
  subnet = "10.56.72.0/24"
  region = "amazon-web-services::us-east-1"
  tags = ["aws"]
}

resource "cloudamqp_instance" "instance" {
  name = "instance"
  plan = "bunny-1"
  region = "amazon-web-services::us-east-1"
  tags = ["aws"]
  vpc_id = cloudamqp_vpc.vpc.id
  keep_associated_vpc = true
}

resource "cloudamqp_security_firewall" "firewall_settings" {
  instance_id = cloudamqp_instance.instance.id
  
  rules {
    ip = "10.56.72.0/24"
    ports = []
    services = ["AMQP","AMQPS"]
  }

  rules {
    ip = "0.0.0.0/0"
    description = "Mgmt interface"
    ports = []
    services = ["HTTPS"]
  }
}
```

## Fetch new identifier

For the basic example above we need to fetch two identifiers to be used when manually update the state file. The identifiers are both the *VPC ID* and *Instance ID*.

> This can be done by using the [CloudAMQP API](https://docs.cloudamqp.com/#list-instances) and listing all the CloudAMQP instances on the new account. From the response, the new CloudAMQP instance have the identifier `209` and the VPC `208`. 

Reviewer: ```suggestion
This can be done by using the [CloudAMQP API](https://docs.cloudamqp.com/#list-instances) and listing all the CloudAMQP instances in the new team. From the response, the new CloudAMQP instance have the identifier `209` and the VPC `208`. 
```
